x-pack/plugins/security_solution/public/management/cypress/tasks/alerts.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { estypes } from '@elastic/elasticsearch';
+import type { Rule } from '../../../detection_engine/rule_management/logic';
+import {
+  DETECTION_ENGINE_QUERY_SIGNALS_URL,
+  DETECTION_ENGINE_RULES_BULK_ACTION,
+  DETECTION_ENGINE_RULES_URL,
+} from '../../../../common/constants';
+import { ELASTIC_SECURITY_RULE_ID } from '../../../../common';
+import { request } from './common';
+import { ENDPOINT_ALERTS_INDEX } from '../../../../scripts/endpoint/common/constants';
+const ES_URL = Cypress.env('ELASTICSEARCH_URL');
+
+/**
+ * Continuously check for any alert to have been received by the given endpoint.
+ *
+ * NOTE:  This is tno the same as the alerts that populate the Alerts list. To check for
+ *        those types of alerts, use `waitForDetectionAlerts()`
+ */
+export const waitForEndpointAlerts = (
```

Comment:
@paul-tavares have you tried to use https://www.npmjs.com/package/cypress-recurse ?

Reply:
No, I have not

Reply:
looks like its available in `package.json` from kibana. Are you proposing that be used instead?

Reply:
Going to merge this in, but if you still have feedback, feel free to post it here on the PR and I'll come back around and follow up on it.